eZ/Publish/Core/Persistence/Legacy/Content/Mapper.php

```diff
@@ -176,64 +181,146 @@ public function convertToStorageValue(Field $field)
      *
      * @param array $rows
      * @param array $nameRows
+     * @param string $prefix
      *
      * @return \eZ\Publish\SPI\Persistence\Content[]
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
      */
-    public function extractContentFromRows(array $rows, array $nameRows, $prefix = 'ezcontentobject_')
-    {
+    public function extractContentFromRows(
+        array $rows,
+        array $nameRows,
+        string $prefix = 'ezcontentobject_'
+    ): array {
         $versionedNameData = [];
+        $languageCodes = [];
+
         foreach ($nameRows as $row) {
-            $contentId = (int)$row['ezcontentobject_name_contentobject_id'];
-            $versionNo = (int)$row['ezcontentobject_name_content_version'];
-            $versionedNameData[$contentId][$versionNo][$row['ezcontentobject_name_content_translation']] = $row['ezcontentobject_name_name'];
+            $contentId = (int)$row["{$prefix}name_contentobject_id"];
+            $versionNo = (int)$row["{$prefix}name_content_version"];
+            $versionedNameData[$contentId][$versionNo][$row["{$prefix}name_content_translation"]] = $row["{$prefix}name_name"];
+            $languageCodes[] = $row["{$prefix}name_content_translation"];
```

Comment:
> `$languageCodes`: Array is only updated, but never queried

AFAIR in previous version it was used later on in the part which I asked to be encapsulated as `loadCachedVersionFieldDefinitionsPerLanguage`. Maybe you can still pass it there and re-use it? if not, then the variable itself is needed.